libdevcore/Log.cpp

```diff
@@ -187,7 +187,7 @@ string dev::getThreadName()
 
 void dev::setThreadName(string const& _n)
 {
-#if defined(__APPLE__)
+#if defined(__GLIBC__) || defined(__APPLE__)
 	pthread_setname_np(_n.c_str());
```

Comment:
Previously there were different pathways for linux and apple. Linux used

```
pthread_setname_np(pthread_self(), _n.c_str());
```

`getname` was the same usage. I guess apple provides implicit `thread_self`.